.github/workflows/test.yml

```diff
@@ -0,0 +1,66 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+    - master
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - python: "3.11"
+          node: 16
+          os: macos-15-intel
+        - python: "3.12"
+          node: 20
+          os: windows-2025
+        - python: "3.13"
+          node: 22
+          os: ubuntu-24.04
+        - python: "3.14"
+          node: 24
+          os: macos-26
+        - python: "3.15-dev"
+          node: 25
+          os: ubuntu-latest
```

Comment:
so why the platforms etc.?
_(was originally trying to find oldest versions; had to use non-latest runners)_. see issue
requirements, cached deps/stack, speed